features/370-auth-ui/design.md

```diff
@@ -0,0 +1,393 @@
+
+### The authorization page
+
+The authorization page is the landing page of the authorize endpoint. It lists out the configured IdPs. It shows a text field for login ID. The login ID field is either a plain text input or a phone number input, depending on the type of the first login ID key.
+
+```
+|---------------------------|
+| Login with Google         |
+|---------------------------|
+| Login with Facebook       |
+|---------------------------|
+
+              Or
+
+|--------------------------------------|  |----------|
+| Enter in your email or username here |  | Continue |
+|--------------------------------------|  |----------|
+
+Login with a phone number instead.
+```
+
+If the user selects an IdP, the user is redirected to the authorization endpoint of that IdP. If the user continues with a login ID, the user is directed to the password page.
+
+Separating the login ID and the password into two pages allow future improvements, such as [Open ID Provider Issuer Discovery](https://openid.net/specs/openid-connect-discovery-1_0.html#IssuerDiscovery). For example, the user can type in their Google account name and then be redirected to the authorization endpoint of Google.
+
+### The signup page
+
+The signup page displays the first login ID key default. Other login ID keys are available to choose.
+
+```
+Sign up with email
+|--------------------------|  |----------|
+| Enter in your email here |  | Continue |
+|--------------------------|  |----------|
+
+Sign up with phone instead.
+Sign up with username instead.
+```
+
+### The signup password page
+
+The signup password page displays a visibility toggleable password field with password requirements.
+
+```
+|------------------------|  |----------|
+| Create a password here |  | Continue |
+|------------------------|  |----------|
+
+- [ ] At least one digit
+- [ ] At least one uppercase English character
+- [ ] At least one lowercase English character
+- [ ] At least one symbols ~`!@#$%^&*()-_=+[{]}\|;:'",<.>/?
+- [ ] At least 8 characters long
+```
+
+## The settings endpoint
+
+The settings endpoint `/_auth/_/settings` allows the user to manage their account. Features include but not limited to:
+
+- Update metadata
+- Change password
+- Configure MFA
+- Add / Update / remove login ID
+- Manage identities
+- Manage sessions
+
+## The verification endpoint
+
+The verification endpoint `/_auth/_/verify` allows the user to verify their login ID. This page requires authenticated user. Unauthenticated access is redirected to the authorization page. Verification is not enforced by the authentication flow. For example, if auto send verification email is enabled, the email is sent behind the scene and the user is redirected to the application immediately after authentication success.
```

Comment:
As mention above, if there are no advantages, maybe we can merge this with the "classic" verify feature?